### PR TITLE
Fix and refactor eth filter methods (part 3)

### PR DESCRIPTION
**Pull request by keroro520**
I have added some test cases, which helped me identify some bugs: https://github.com/nervosnetwork/godwoken-tests/pull/143

This PR brings some small bugfixes for them, and some necessary refactoring.

----

This PR is based on https://github.com/nervosnetwork/godwoken-web3/pull/427 and https://github.com/nervosnetwork/godwoken-web3/pull/428

### CODE REVIEW

File: packages/api-server/src/methods/modules/eth.ts [hunk 1]
```diff
@@ -1230,6 +1097,57 @@ export class Eth {
...
+      blockHash: filter.blockHash,
+    };
+  }
+
+  private async _normalizeBlockParameterForFilterRequest(
+    blockParameter: undefined | BlockParameter
+  ): Promise<bigint> {
+    const latestBlockNumber = await this.getTipNumber();
+    if (
+      blockParameter == undefined ||
```
Comment (classicalliu) [inline packages/api-server/src/methods/modules/eth.ts:1138]:
```suggestion
      blockParameter == null ||
```

**Review by classicalliu — APPROVED** [review]

**Comment by RetricSu** [comment]
please rebase

**Comment by keroro520** [comment]
@RetricSu @classicalliu Rebased. PTAL.
CI fails because of another issue.

File: packages/api-server/src/db/helpers.ts [hunk 2]
```diff
@@ -223,75 +221,21 @@ export function filterLogsByAddress(
...
-export interface ExecuteOneQueryResult {
-  status: QueryRoundStatus;
-  data: any[];
-}
-
 export function getDatabaseRateLimitingConfiguration() {
   const MAX_QUERY_NUMBER = envConfig["maxQueryNumber"]
     ? +envConfig["maxQueryNumber"]
     : DEFAULT_MAX_QUERY_NUMBER;
+  // TODO set query timeout
```
Comment (RetricSu) [inline packages/api-server/src/db/helpers.ts:228]: can we add query time limit in this pr as well? this is important after db grows. I think simple solution like `Promise.race([query, timeout])` should be enough?

Comment (sopium) [reply packages/api-server/src/db/helpers.ts:228]: Knex has a `timeout` method:

http://knexjs.org/guide/query-builder.html#timeout

If it actually cancels the query then it's better than `Promise.race`.

Comment (RetricSu) [reply packages/api-server/src/db/helpers.ts:228]: > http://knexjs.org/guide/query-builder.html#timeout

cool this is more clean

File: packages/api-server/src/db/helpers.ts [hunk 3]
```diff
@@ -223,75 +221,21 @@ export function filterLogsByAddress(
...
   const MAX_QUERY_TIME_MILSECS = envConfig["maxQueryTimeInMilliseconds"]
     ? +envConfig["maxQueryTimeInMilliseconds"]
     : DEFAULT_MAX_QUERY_TIME_MILSECS;
-  const MAX_QUERY_ROUNDS = envConfig["maxQueryRounds"]
```
Comment (RetricSu) [inline packages/api-server/src/db/helpers.ts:243]: since maxQueryRounds is removed, please update `env-config.ts` and `README.md` as well

**Review by RetricSu — CHANGES_REQUESTED** [review]

File: packages/api-server/src/db/helpers.ts (hunk 2 above)
Comment (keroro520) [reply packages/api-server/src/db/helpers.ts]: Thanks, @sopium @RetricSu 

The new commit f7771bbd653765382aff6dd3cc0fa833258b32d8 uses http://knexjs.org/guide/query-builder.html#timeout to add query  timeout limit. PTAL.

**Review by RetricSu — APPROVED** [review]

**Pull request merged**